aws-s3-transfer-manager/tests/download_test.rs

```diff
@@ -148,26 +156,43 @@ async fn test_download_ranges() {
         requests[2].headers().get("Range"),
         Some("bytes=10485760-12582911")
     );
-
-    handle.join().await.unwrap();
 }
 
-/// Test body not consumed which should not prevent the handle from being joined
+/// Test output not consumed which should not prevent the handle from being dropped
 #[tokio::test]
-async fn test_body_not_consumed() {
+async fn test_output_not_consumed() {
     let data = rand_data(12 * MEBIBYTE);
     let part_size = 5 * MEBIBYTE;
 
     let (tm, _) = simple_test_tm(&data, part_size);
 
-    let handle = tm
+    let mut handle = tm
         .download()
         .bucket("test-bucket")
         .key("test-object")
         .initiate()
         .unwrap();
 
-    handle.join().await.unwrap();
+    let _ = handle.ouput_mut().next().await;
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_abort_download() {
```

Comment:
You don't have to necessarily use it here but one thing to make you aware of is you can leverage `capture_test_logs` ([example](https://github.com/awslabs/aws-s3-transfer-manager-rs/blob/cancel-single-download/aws-s3-transfer-manager/tests/upload_test.rs#L117)) which you can use to assert the logs contain certain things (e.g. to verify particular branches were hit).